### PR TITLE
aditi.maheshwari/telemetry-log-metric-genration-test

### PR DESCRIPTION
**Pull request by aditim16**
## Description

Assert that app-generate-metrics is not sent when DD_TELEMETRY_METRICS_COLLECTION_ENABLED, DD_TELEMETRY_LOGS_COLLECTION_ENABLED is false
## Workflow

1. ⚠️⚠️ Create your PR as draft (we're receiving lot of PR, it saves us lot of time) ⚠️⚠️
2. Follows the style guidelines of this project (See [how to easily lint the code](https://github.com/DataDog/system-tests/blob/main/docs/edit/lint.md))
3. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
4. Mark it as ready for review

> **_NOTE:_**  By default in PR only default scenario tests will be launched. Please refer to the [documentation](https://datadoghq.atlassian.net/wiki/spaces/APMINT/pages/2866381467/CI+Workflow+Github+Actions) to run all scenarios in your PR if needed.

Once your PR is reviewed, you can merge it ! :heart:

### CODE REVIEW

**Review by cbeauchesne — APPROVED** [review]

File: tests/test_telemetry.py [hunk 1]
```diff
@@ -409,3 +409,37 @@ def test_message_batch_event_order(self):
             < eventslist.index("app-integrations-change")
             < eventslist.index("app-product-change")
         ), "Events in message-batch are not in chronological order of event triggered"
+
+
+@released(cpp="?", dotnet="?", golang="?", java="?", nodejs="?", php="?", python="?", ruby="?")
+@scenarios.telemetry_log_generation_disabled
+class Test_Log_Genration:
```
Comment (mehulsonowal) [inline tests/test_telemetry.py:416]:
```suggestion
class Test_Log_Generation:
```

File: tests/test_telemetry.py [hunk 2]
```diff
@@ -409,3 +409,37 @@ def test_message_batch_event_order(self):
...
+    """Assert that logs are not reported when logs generation is disabled in telemetry"""
+
+    def test_log_genration_disabled(self):
```
Comment (mehulsonowal) [inline tests/test_telemetry.py:419]:
```suggestion
    def test_log_generation_disabled(self):
```

File: tests/test_telemetry.py [hunk 3]
```diff
@@ -409,3 +409,37 @@ def test_message_batch_event_order(self):
...
+                content = data["request"]["content"]
+                raise Exception(" Logs event is sent when log generation is disabled")
+
+
+@released(cpp="?", dotnet="?", golang="?", java="?", nodejs="?", php="?", python="?", ruby="?")
+@scenarios.telemetry_metric_generation_disabled
+class Test_Metric_Generation:
+    """Assert that metrics are not reported when metric generation is disabled in telemetry"""
+
+    def test_metric_genration_disabled(self):
```
Comment (mehulsonowal) [inline tests/test_telemetry.py:436]:
```suggestion
    def test_metric_generation_disabled(self):
```

**Review by mehulsonowal — APPROVED** [review]
nits on typos but LGTM

File: tests/test_telemetry.py (hunk 1 above)
Comment (cbeauchesne) [reply tests/test_telemetry.py]: Actually,  `Test_LogGeneration`

**Pull request merged**